### PR TITLE
remove comparisons of nonnull objects against 0

### PR DESCRIPTION
**Pull request by davidlange6**
indicated by warnings in gcc630

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: RecoTracker/SingleTrackPattern/src/CRackTrajectoryBuilder.cc [hunk 1]
```diff
@@ -412,152 +412,89 @@ CRackTrajectoryBuilder::SortHits(const SiStripRecHit2DCollection &collstereo,
...
-	  if ( noSeedDet )
+      for(istripm=collmatched.data().begin();istripm!=collmatched.data().end();istripm++){
+	float ych= RHBuilder->build(&(*istripm))->globalPosition().y();
+	
+	int cDetId=istripm->geographicalId().rawId();
+	bool noSeedDet = ( detIDSeedMatched.end() == find (detIDSeedMatched.begin(), detIDSeedMatched.end(), cDetId ) ) ;
+	
+	if ( noSeedDet )
 	  if ((seed_plus && (ych<yref)) || (!(seed_plus) && (ych>yref)))
-	    {
```
Comment (slava77) [inline RecoTracker/SingleTrackPattern/src/CRackTrajectoryBuilder.cc]: I'm not a big fan of dropping braces for one-line if/else.
It's easy to get things pretty wrong this way if any edits are to be made around these lines

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17364	11c009b
- technical (removing equivalent of "if (true)" )
- jenkins tests pass

**Comment by davidlange6** [comment]
+1

**Pull request merged**